python/cugraph/comms/comms.py

```diff
@@ -14,15 +14,65 @@
 from cugraph.raft.dask.common.comms import Comms as raftComms
 from cugraph.raft.dask.common.comms import worker_state
 from cugraph.raft.common.handle import Handle
+from cugraph.comms.comms_wrapper import init_subcomms as c_init_subcomms
+from dask.distributed import default_client
+import math
 
 
 __instance = None
 __default_handle = None
+__subcomm = None
+
+
+def __get_2D_div(ngpus):
+    pcols = int(math.sqrt(ngpus))
+    while ngpus % pcols != 0:
+        pcols = pcols - 1
+    return int(ngpus/pcols), pcols
+
+
+def subcomm_init(prows, pcols, partition_type):
+    sID = get_session_id()
+    ngpus = get_n_workers()
+    if prows is None and pcols is None:
+        if partition_type == 1:
+            pcols, prows = __get_2D_div(ngpus)
+        else:
+            prows, pcols = __get_2D_div(ngpus)
+    else:
+        if prows is not None and pcols is not None:
+            if ngpus != prows*pcols:
+                raise Exception('prows*pcols should be equal to the\
+ number of processes')
+        elif prows is not None:
+            if ngpus % prows != 0:
+                raise Exception('prows must be a factor of the number\
+ of processes')
+            pcols = int(ngpus/prows)
+        elif pcols is not None:
+            if ngpus % pcols != 0:
+                raise Exception('pcols must be a factor of the number\
+ of processes')
+            prows = int(ngpus/pcols)
+
+    client = default_client()
+    client.run(_subcomm_init, sID, pcols)
+    global __subcomm
+    __subcomm = (prows, pcols, partition_type)
+
+
+def _subcomm_init(sID, partition_row_size):
+    handle = get_handle(sID)
+    c_init_subcomms(handle, partition_row_size)
 
 
 # Intialize Comms. If explicit Comms not provided as arg,
 # default Comms are initialized as per client information.
```

Comment:
As this is a public interface, don't we need to update the documentation?

Reply:
And we eventually need to remove the C++ partition manager class. This class is basically a misnomer... C++ is not managing partitioning. The python layer is managing partitioning and the C++ layer is just performing a passive task based on instructions from the python layer.

Reply:
Yes, thank you. I'll work on that now.